### PR TITLE
Add TLS support to the Trillian log/map servers

### PR DESCRIPTION
**Pull request by maljub01**
Addresses #428

### CODE REVIEW

File: server/main.go [hunk 1]
```diff
@@ -53,9 +56,15 @@ type Main struct {
 	// Endpoints for RPC and HTTP/REST servers.
 	// HTTP/REST is optional, if empty it'll not be bound.
 	RPCEndpoint, HTTPEndpoint string
-	DB                        *sql.DB
-	Registry                  extension.Registry
-	Server                    *grpc.Server
+
+	// TLS Certificate and Key files for the server.
+	TlsCertFile, TlsKeyFile string
```
Comment (AlCutter) [inline server/main.go]: `Tls`->`TLS`

File: server/main.go [hunk 2]
```diff
@@ -137,6 +165,31 @@ func (m *Main) Run(ctx context.Context) error {
 	return nil
 }
 
+// newGrpcServer starts a new Trillian gRPC server.
+func (m *Main) newGrpcServer() (*grpc.Server, error) {
```
Comment (AlCutter) [inline server/main.go]: `Grpc`->`GRPC`

File: server/trillian_log_server/main.go [hunk 3]
```diff
@@ -60,6 +58,8 @@ var (
 	mySQLURI        = flag.String("mysql_uri", "test:zaphod@tcp(127.0.0.1:3306)/test", "Connection URI for MySQL database")
 	rpcEndpoint     = flag.String("rpc_endpoint", "localhost:8090", "Endpoint for RPC requests (host:port)")
 	httpEndpoint    = flag.String("http_endpoint", "localhost:8091", "Endpoint for HTTP metrics and REST requests on (host:port, empty means disabled)")
+	tlsCertFile     = flag.String("tls_cert_file", "", "Path to the TLS server certificate.")
```
Comment (AlCutter) [inline server/trillian_log_server/main.go]: Consider adding a bit more text to the description here too, about running a non-TLS server if you don't specify these flags.

File: server/trillian_log_server/main.go [hunk 4]
```diff
@@ -135,20 +135,15 @@ func main() {
...
-	ti := interceptor.New(
-		registry.AdminStorage, registry.QuotaManager, *quotaDryRun, registry.MetricFactory)
-	netInterceptor := interceptor.Combine(stats.Interceptor(), interceptor.ErrorWrapper, ti.UnaryInterceptor)
-	s := grpc.NewServer(grpc.UnaryInterceptor(netInterceptor))
-	// No defer: server ownership is delegated to server.Main
-
 	m := server.Main{
 		RPCEndpoint:  *rpcEndpoint,
 		HTTPEndpoint: *httpEndpoint,
+		TlsCertFile:  *tlsCertFile,
```
Comment (AlCutter) [inline server/trillian_log_server/main.go]: `Tls` -> `TLS`

File: server/trillian_map_server/main.go [hunk 5]
```diff
@@ -58,6 +55,8 @@ var (
 	mySQLURI     = flag.String("mysql_uri", "test:zaphod@tcp(127.0.0.1:3306)/test", "Connection URI for MySQL database")
 	rpcEndpoint  = flag.String("rpc_endpoint", "localhost:8090", "Endpoint for RPC requests (host:port)")
 	httpEndpoint = flag.String("http_endpoint", "localhost:8091", "Endpoint for HTTP metrics and REST requests on (host:port, empty means disabled)")
+	tlsCertFile  = flag.String("tls_cert_file", "", "Path to the TLS server certificate.")
```
Comment (AlCutter) [inline server/trillian_map_server/main.go]: (here too)

Comment (maljub01) [reply server/trillian_map_server/main.go]: Done.

File: server/trillian_log_server/main.go (hunk 4 above)
Comment (maljub01) [reply server/trillian_log_server/main.go]: Done.

File: server/trillian_log_server/main.go (hunk 3 above)
Comment (maljub01) [reply server/trillian_log_server/main.go]: Done.

File: server/main.go (hunk 2 above)
Comment (maljub01) [reply server/main.go]: Done.

File: server/main.go (hunk 1 above)
Comment (maljub01) [reply server/main.go]: Done.

File: server/trillian_log_server/main.go [hunk 6]
```diff
@@ -58,8 +58,8 @@ var (
 	mySQLURI        = flag.String("mysql_uri", "test:zaphod@tcp(127.0.0.1:3306)/test", "Connection URI for MySQL database")
 	rpcEndpoint     = flag.String("rpc_endpoint", "localhost:8090", "Endpoint for RPC requests (host:port)")
 	httpEndpoint    = flag.String("http_endpoint", "localhost:8091", "Endpoint for HTTP metrics and REST requests on (host:port, empty means disabled)")
-	tlsCertFile     = flag.String("tls_cert_file", "", "Path to the TLS server certificate.")
-	tlsKeyFile      = flag.String("tls_key_file", "", "Path to the TLS server key.")
+	tlsCertFile     = flag.String("tls_cert_file", "", "Path to the TLS server certificate. If not set, the server will use unsecured connections.")
```
Comment (AlCutter) [inline server/trillian_log_server/main.go]: "not set -> unset" * 4 :)

**Pull request merged**